### PR TITLE
Bump k8s v1.8.1

### PR DESCRIPTION
**Pull request by aaronlevy**
I'll wait on bumping vendor libs to v1.8 until client-go has a more unified v1.8 targeted release.

cc @rphillips @dghubble

### CODE REVIEW

**Comment by aaronlevy** [comment]
Also, running conformance tests separately - will hold off on merging until those pass too

**Comment by rphillips** [comment]
lgtm

**Comment by dghubble** [comment]
Tested PXE booted bare-metal setups with both `flannel` and with `calico`. Bootstrapping succeeds and control plane seems ok (nothing too atypical in logs). This didn't cover controller reboot survival (i.e. checkpointing).

**Comment by aaronlevy** [comment]
Conformance has 2 failures need to check out:

```
[1061461.011458] golang[5]: [14168.336421] golang[5]: Summarizing 2 Failures:
[1061461.011722] golang[5]: [14168.336603] golang[5]: [Fail] [k8s.io] Pods [It] should be submitted and removed [Conformance] 
[1061461.012043] golang[5]: [14168.336787] golang[5]: /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/pods.go:252
[1061461.012319] golang[5]: [14168.336972] golang[5]: [Fail] [k8s.io] Pods Extended [k8s.io] Delete Grace Period [It] should be submitted and removed [Conformance] [Flaky]
```

@dghubble I think we might have to bump the vendoring after all (or update the tests). Not super clear about the interaction of using "extensions/v1beta1" in the tests, vs creating the objects with v1.8 "apps/v1beta2"

**Comment by dghubble** [comment]
Any idea if my suggestion will unblock the e2e tests, might be worth a shot. I hadn't looked at conformance tests yet.

**Comment by aaronlevy** [comment]
I believe the intention there is to remove the nodeSelector altogether (so the daemonset ends up on all nodes) - we're not using node affinity in this case. I have a feeling the actual failure is due to some versioning interplay (e.g. using `ExtensionsV1beta1()` when it's been switched to apps/v1beta2.

**Comment by aaronlevy** [comment]
coreosbot run e2e

**Comment by aaronlevy** [comment]
@dghubble do you know how to re-trigger the e2e-calico test?

**Comment by dghubble** [comment]
coreosbot run e2e calico

**Comment by dghubble** [comment]
Happy to change it, its just the first phrase I defined.

**Comment by aaronlevy** [comment]
@dghubble that's fine. Mind adding it to https://github.com/kubernetes-incubator/bootkube/blob/master/Documentation/development.md#running-pr-tests

**Comment by dghubble** [comment]
coreosbot run e2e calico

**Comment by dghubble** [comment]
e2e-calico failed bc Terraform is waiting for an EC2 instance to create, retry.

**Comment by dghubble** [comment]
Both failures are related to the checkpointer. Any relation to the (not-yet-included) improvements made in #733 @thorfour?

**Comment by dghubble** [comment]
@aaronlevy I see e2e and self-hosted etcd e2e are using "coreosbot run e2e", so I've changed calico-e2e to use the same. Docs are already correct.

**Comment by thorfour** [comment]
My changes wouldn't be a fix for those failures. I believe I've seen those failures before, seems like a timing failure

**Comment by aaronlevy** [comment]
I don't think it's related to checkpointer itself, but rather behavior that touches the checkpointer tests. It seems like there's something related to removing pods that has changed (similarly got same conformance test failures twice now - so not outright flakes).

**Comment by dghubble** [comment]
Looking at the tests, we're rebooting masters, starting workers, verifying "pod-checkpointer" is present on a worker (`verifyPod` with `docker ps`), and starting masters, and then verifying "pod-checkpointer" is NOT present on a worker (step that fails).
https://github.com/kubernetes-incubator/bootkube/blame/master/e2e/checkpointer_test.go#L277
https://github.com/kubernetes-incubator/bootkube/blame/master/e2e/checkpointer_test.go#L89

I'm guessing this is done to test the idea that once an apiserver can be contacted, the checkpointer doesn't need to be running on that worker per [doc](https://github.com/kubernetes-incubator/bootkube/blob/master/cmd/checkpoint/README.md#description), but I'm not 100%. cc @yifan-gu

**Comment by aaronlevy** [comment]
Added a sleep and the tests pass. I have a feeling removing a pod might have got slower? One option might just be to increase the retry window on `verifyPod`

**Comment by dghubble** [comment]
Increasing the retries allowed does seem nicer than a fixed sleep.

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -161,7 +169,7 @@ spec:
         - /var/lock/api-server.lock
         - /hyperkube
         - apiserver
-        - --admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,ResourceQuota
+        - --admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds
```
Comment (s-urbaniak) [inline pkg/asset/internal/templates.go]: quick question: do we have to take over these `admission-control` setting changes also in the Tectonic installer?

Comment (aaronlevy) [reply pkg/asset/internal/templates.go]: I cc'd you on an internal issue and will send a follow up email.

**Comment by dghubble** [comment]
Looks like that cluster apiserver failed to come back and subsequent tests failed. Hm, known or new?

**Comment by aaronlevy** [comment]
The self-hosted etcd is actually still occasionally flaky - and not really maintained. I'll re-run for now (last test with the sleep passed, and this is functionally the same change).

**Comment by aaronlevy** [comment]
coreosbot run e2e

**Comment by aaronlevy** [comment]
I'm going to try and see if I can reproduce locally. Even if this does pass - having to run e2e test 3 times is too flaky

**Comment by dghubble** [comment]
Same flake waiting for checkpointer to restore apiserver in the calico variant, so it appears independent of just self-hosted etcd.

**Comment by aaronlevy** [comment]
I feel like something about deleting pods has slowed way down

**Comment by dghubble** [comment]
I recall a discussion about pod deletion slowdowns. cc @ericchiang similar or something else? Perhaps initial test assumptions for delete and wait times no longer hold true.

**Comment by rphillips** [comment]
There is an issue fixed in 1.8.1 (not released yet) that was specifically for deleting pods [#51899](https://github.com/kubernetes/kubernetes/issues/51899)

**Comment by dghubble** [comment]
Hm, the description of large clusters doesn't seem to match our much smaller test clusters.

**Comment by rphillips** [comment]
True... The [fix](https://github.com/kubernetes/kubernetes/pull/53233) states the periodic garbage collector would need to run and instead they have changed the code to actively remove the pods.

**Comment by aaronlevy** [comment]
But waiting up to 30 seconds to delete a pod might add to this flakiness...

We could add longer retry timeouts until v1.8.1 then back them out again.

**Comment by aaronlevy** [comment]
@dghubble @rphillips - passing at this point. Seem reasonable to merge?

I have a strong suspicion that the two conformance tests failures are also timing related. I manually tested deleting pods (those tests are essentially checking the grace period) and didn't see any problems. We should still did into those errors and get them resolved, but I feel like it's relatively low risk here.

**Comment by rphillips** [comment]
lgtm, 1.8.1 released today fwiw

**Comment by ericchiang** [comment]
I think https://github.com/kubernetes/kubernetes/issues/53396 (hyperkube v1.8.0 breaks pods talking to services on the same node) still impacts us.

**Comment by dghubble** [comment]
@aaronlevy I haven't reproduced bootstrapping flakes testing this manually. Can we aim to bump to 1.8.1 and patch the hyperkube before a bootkube release? I'm ok with merging or if you want to wait for that followup.

**Comment by aaronlevy** [comment]
Hmm https://github.com/kubernetes/kubernetes/issues/53396 is bad enough that we should just bump the image.

@dghubble is anyone working on bumping the hyperkube?

**Comment by rphillips** [comment]
@aaronlevy the patch for #53396 hasn't been merged yet, but it should be shortly (probably today)

**Comment by ericchiang** [comment]
The fix for that issue was submitted too late to get into 1.8.1

https://github.com/kubernetes/kubernetes/pull/53642

@rphillips opened a cherry pick which can be used to track the fix for the
branch

https://github.com/kubernetes/kubernetes/pull/53744

If we're going to release another hyperkube, we'll probably want to cherry
pick those changes once the original fix is merged into master.

**Comment by dghubble** [comment]
Ok if we want to wait until all the pieces are ready to go at once.

**Comment by squat** [comment]
@aaronlevy I started building v1.8.1 this morning, tag is cut. Waiting for build to finish and image push to finish.

**Comment by aaronlevy** [comment]
Heh, whoops. I'll revert the v1.8.1 bump then

**Comment by aaronlevy** [comment]
Per some tests by @rphillips - the networking issue has to do with kubenet - so I think we can move forward with merging this now. hyperkube v1.8.1 is also available, so I bumped that. If someone gets a chance please do another review and I'll merge on green tests

**Comment by dghubble** [comment]
coreosbot run e2e

**Comment by aaronlevy** [comment]
@dghubble any chance you could do a quick manual sanity check of calico + the reboot test that it says failed?

**Comment by dghubble** [comment]
Yeah, I'm working on testing this on some clusters.

**Comment by dghubble** [comment]
On bare-metal with `bootkube render` at this commit and `--network-provider=experimental-calico`, controller and full cluster reboots were tolerated as usual.

```
NAMESPACE     NAME                                       READY     STATUS    RESTARTS   AGE
kube-system   calico-node-c6596                          2/2       Running   0          8m
kube-system   calico-node-j2kdj                          2/2       Running   2          8m
kube-system   calico-node-w65hg                          2/2       Running   0          8m
kube-system   kube-apiserver-55ctg                       1/1       Running   1          8m
kube-system   kube-controller-manager-5dbc5d8c6b-bd2v5   1/1       Running   1          8m
kube-system   kube-controller-manager-5dbc5d8c6b-cmbh9   1/1       Running   1          8m
kube-system   kube-dns-598c789574-9vgnd                  3/3       Running   3          7m
kube-system   kube-proxy-8nwbt                           1/1       Running   1          7m
kube-system   kube-proxy-c8x54                           1/1       Running   0          7m
kube-system   kube-proxy-jgfg8                           1/1       Running   0          7m
kube-system   kube-scheduler-bdb68cdc-4tqzp              1/1       Running   1          7m
kube-system   kube-scheduler-bdb68cdc-5qhp5              1/1       Running   1          7m
kube-system   pod-checkpointer-85p29                     1/1       Running   1          7m
kube-system   pod-checkpointer-85p29-node1.example.com   1/1       Running   1          7m
```

rel: https://github.com/coreos/matchbox/pull/657

**Comment by dghubble** [comment]
On CI, TestReboot failed once and TestCheckpointerUnscheduleCheckpointer failed once on separate runs. So its a flaking issue rather than being broken outright.

I'll point out, the reboot seems to fail sometimes at the testing code's call to node.Reboot, not from a cluster behavior itself. `reboot_test.go:30: failed to reboot node: Process exited with status 1`

**Comment by dghubble** [comment]
Tested on a cloud provider as well

**Comment by dghubble** [comment]
coreosbot run e2e

**Comment by dghubble** [comment]
e2e-selfetcd failed on Terraform creation of AWS resources. `aws_instance.master_node.0: Error waiting for instance (i-0c6b293b70086d5fa) to become ready: Failed to reach target state. Reason: Server.InternalError: Internal error on launch`

**Comment by dghubble** [comment]
coreosbot run e2e

**Comment by aaronlevy** [comment]
Any thoughts on how we might make this less flaky?

**Comment by dghubble** [comment]
* TestReboot failures sometimes occur simply because the reboot logic is literally try to ssh once and run "reboot". No retries.
* Most recent nginx workload failure comes from https://github.com/coreos/ktestutil. Hasn't been touched in a while, old client-go? It hardcodes its own retry intervals and counts, so they're unaffected by your bumps. https://github.com/coreos/ktestutil/tree/master/testworkload

At a higher level and longer term level, I'd like to strip out functionality we don't intend to go forward with such as self-hosted kubelet (#735, but not an issue here) and self-hosted etcd. Most of these issues boil down to crash loops being a normal part of self-hosted control planes, our control plane growing more complex, and there being no precise definition of when setup is complete and conformance testing should begin. It'd also be valuable if we had some better way to show how long a step normally takes, we kinda go by instinct and maybe these values need re-tuning.

**Comment by dghubble** [comment]
coreosbot run e2e

**Comment by aaronlevy** [comment]
I'm really hesitant to merge this even if tests finally pass (without addressing some of the underlying issues). Otherwise every subsequent PR will have to run tests 5 times to pass (which isn't the case prior to this PR). So that might mean that we need to increase timeouts.

**Comment by dghubble** [comment]
Agreed. At this point its mainly useful to see where we're hitting failures. It'd be different if it were the same error every time.

**Comment by klausenbusk** [comment]
> I'd like to strip out functionality we don't intend to go forward with such as self-hosted kubelet (#735, but not an issue here) and **self-hosted etcd**

Sorry for hijacking this issue, but why do you want to drop self-hosted etcd? Isn't it working? Too unstable? I can see it was just [removed](https://github.com/coreos/tectonic-installer/pull/1971) from the tectonic UI.

**Comment by dghubble** [comment]
This isn't the place for that discussion, it would hijack this. I'll open an issue later to communicate the plan more broadly.

**Comment by dghubble** [comment]
Summarizing here, in the 5 builds on v1.8.1 (rather than v1.8.0)
* e2e - 5/5 passes
* e2e-calico - 4/5 passes
  * One failure on TestReboot due to ssh'ing once and not retrying (known test issue)
* e2e-selfetcd - 2/5 passes
  * One failure due to Terraform AWS failing to create instance (known flaky)
  * One failure on TestSmoke waiting for nginx creation (unknown)
  * One failure on TestDeleteAPI and subsequent tests "non-checkpoint apiserver did not come back". (unknown)

**Comment by aaronlevy** [comment]
All conformance tests passed once bumped to v1.8.1

**Comment by dghubble** [comment]
Added some fixes which should address the TestReboot and TestSmoke flakes. I don't have anything for the Terraform or occassional self-hosted etcd TestDeleteAPI failures. I haven't been able to reproduce them on my clusters running v1.8.1 with self-hosted etcd.

**Comment by dghubble** [comment]
Ok, so we've seen flaky failures in TestDeleteAPI "non-checkpoint apiserver did not come back" on both e2e-calico and e2e-selfetcd

**Comment by aaronlevy** [comment]
Looks like all passed. Good to merge @dghubble ?

**Comment by dghubble** [comment]
I'll keep trying to repro the flakes in experimental features.

Yes, good to merge.

**Pull request merged**